src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -248,6 +248,11 @@ static void checkAttributes(AnonymousFunctionExpressionSyntax syntax, SyntaxList
             MessageID.IDS_FeatureLambdaReturnType.CheckFeatureAvailability(diagnostics, syntax);
 
             syntax = syntax.SkipRef(out RefKind refKind);
+            if ((syntax as IdentifierNameSyntax)?.Identifier.ContextualKind() == SyntaxKind.VarKeyword)
+            {
+                diagnostics.Add(ErrorCode.ERR_LambdaExplicitReturnTypeVar, syntax.Location);
```

Comment:
Probably should document as breaking change in 17.1

Reply:
Does the spec capture that design decision?

Reply:
I've added an item to the breaking changes document. I'll update the spec in a subsequent PR.